suite-native/notifications/src/components/ToastNotification.tsx

```diff
@@ -46,54 +47,84 @@ const IconContainerStyle = prepareNativeStyle<{
     ),
 }));
 
-const ToastNotificationVariantToStyleMap = {
+const ToastTypeToContentMap = {
+    'copy-to-clipboard': {
+        variant: 'default',
+        iconName: 'copy',
+        defaultMessage: 'Copied to clipboard',
+    },
+    'graph-error': {
+        variant: 'warning',
+        iconName: 'warningCircle',
+        defaultMessage: 'Error loading the graph',
+    },
+    'redirect-error': {
+        variant: 'error',
+        iconName: 'warningTriangle',
+        defaultMessage: 'Unable to open the link',
+    },
+} as const satisfies Record<
+    NativeToastType,
+    { variant: ToastNotificationVariant; iconName: IconName; defaultMessage: string }
+>;
+// todo change to toastNative tipe
+
+const ToastVariantToStyleMap = {
     default: {
         backgroundColor: 'backgroundNeutralBold',
         iconBackgroundColor: 'iconSubdued',
         textColor: 'textOnPrimary',
         iconColor: 'iconOnPrimary',
-        defaultIcon: 'copy',
     },
     success: {
         backgroundColor: 'backgroundPrimarySubtleOnElevation0',
         iconBackgroundColor: 'backgroundPrimarySubtleOnElevation1',
         textColor: 'textPrimaryDefault',
         iconColor: 'iconPrimaryDefault',
-        defaultIcon: 'check',
     },
     warning: {
         backgroundColor: 'backgroundAlertYellowSubtleOnElevation0',
         iconBackgroundColor: 'backgroundAlertYellowSubtleOnElevation1',
         textColor: 'textAlertYellow',
         iconColor: 'iconAlertYellow',
-        defaultIcon: 'warningCircle',
     },
     error: {
         backgroundColor: 'backgroundAlertRedSubtleOnElevation0',
         iconBackgroundColor: 'backgroundAlertRedSubtleOnElevation1',
         textColor: 'textAlertRed',
         iconColor: 'iconAlertRed',
-        defaultIcon: 'warningTriangle',
     },
     info: {
         backgroundColor: 'backgroundAlertBlueSubtleOnElevation0',
         iconBackgroundColor: 'backgroundAlertBlueSubtleOnElevation1',
         textColor: 'textAlertBlue',
         iconColor: 'iconAlertBlue',
-        defaultIcon: 'info',
     },
 } as const satisfies Record<ToastNotificationVariant, ToastNotificationStyle>;
 
-export const ToastNotification = ({ variant = 'default', title, icon }: ToastNotificationProps) => {
+export const ToastNotification = memo(({ type, message }: ToastNotificationProps) => {
```

Comment:
Does this really need to be memoized or are you just using this "to be safe"?

Reply:
It does, if it's not memoized, then the timeout of hiding the toast is refreshed and takes longer to disappear then designed.

Reply:
Oh, i missed that, you ar eprobably right. I will try to remove memo and add useEffect deps array.

Reply:
fixed ce38fec3e